### PR TITLE
Load tables dynamically in Elasticsearch

### PR DESCRIPTION
**Pull request by martint**
It's no longer necessary to have a static configuration to declare which tables are available and their schema. They are now derived from the metadata exposed by ES.

The following data types are now supported: `tinyint`, `smallint`, `integer`, `real`, `binary`, `timestamp`. Additionally, the `_id`, `_score` and `_source` (hidden) columns are also available for querying.

### CODE REVIEW

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/IntegerDecoder.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+{
+    @Override
+    public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
+    {
+        Object value = getter.get();
+        if (value == null) {
+            output.appendNull();
+        }
+        else {
+            INTEGER.writeLong(output, ((Number) value).intValue());
```
Comment (dain) [inline presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/IntegerDecoder.java]: I would use the exact functions for all of these

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchClient.java [hunk 2]
```diff
@@ -297,65 +315,154 @@ private int shardPreference(SearchShardsResponse.Shard left, SearchShardsRespons
...
+            JsonNode value = field.getValue();
+            if (value.has("type")) {
+                String type = value.get("type").asText();
+
+                if (type.equals("date")) {
+                    List<String> formats = ImmutableList.of();
+                    if (value.has("format")) {
+                        formats = Arrays.asList(value.get("format").asText().split("\\|\\|"));
+                    }
+                    result.add(new IndexMetadata.Field(name, new IndexMetadata.DateTimeType(name, formats)));
```
[the remark this answers is not in the document]
Comment (martint) [reply presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchClient.java]: Nope, left-over from an earlier version.

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchPageSource.java [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+            currentPosition = 0;
+        }
+
+        public void close()
+        {
+            if (scrollId != null) {
+                try {
+                    client.clearScroll(scrollId);
+                }
+                catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (martint) [reply presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchPageSource.java]: I don't want queries to fail if something happens during cleanup. The scroll will expire and get cleaned up automatically, anyway.
I'll add logging, though.

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/RowDecoder.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    public RowDecoder(List<String> fieldNames, List<Decoder> decoders)
+    {
+        this.fieldNames = fieldNames;
+        this.decoders = decoders;
+    }
+
+    @Override
+    public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
+    {
+        Map<String, Object> data = (Map<String, Object>) getter.get();
```
[the remark this answers is not in the document]
Comment (martint) [reply presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/RowDecoder.java]: Yes, unless ES returns something unexpected. I'm adding a check here, too.

**Pull request merged**